internal/pkg/deploy/cloudformation/app_test.go

```diff
@@ -1204,3 +1205,525 @@ func TestCloudFormation_RenderStackSet(t *testing.T) {
+				cfn.EXPECT().Describe(stack.NameForAppStack("phonetool")).Return(&cloudformation.StackDescription{
+					Parameters: []*awscfn.Parameter{
+						{
+							ParameterKey:   aws.String("AppDNSDelegatedAccounts"),
+							ParameterValue: aws.String("1234,5678"),
+						},
+					},
+				}, nil)
+				cfn.EXPECT().UpdateAndWait(gomock.Any())
+				appStackSet.EXPECT().Describe("phonetool-infrastructure").Return(stackset.Description{
+					ID:   "",
+					Name: "phonetool-infrastructure",
+					Template: `Metadata:
+  TemplateVersion: 'v1.2.0'
+  Version: 17
+  Workloads:
+    - Name: ar
+      WithECR: true
+  Accounts:
+    - 1234
+    - 5678`,
+				}, nil)
+				appStackSet.EXPECT().Update("phonetool-infrastructure", gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return("123", nil)
+
+				return CloudFormation{
+					cfnClient: cfn,
+					region:    "us-east-1",
+
+					appStackSet: appStackSet,
+					dnsDelegatedAccountsForStack: func(in *awscfn.Stack) []string {
+						return []string{"1234", "5678"}
+					},
+					renderStackSet: func(in renderStackSetInput) error {
+						_, err := in.createOpFn()
+						return err
+					},
+					s3Client: s3,
+					regionalECRClient: func(region string) imageRemover {
+						return ecr
+					},
+					regionalClient: func(region string) cfnClient { return regionalCfn },
+				}
+			},
+		},
+		"skips stack redeployment if 'RemoveAccountFromApp' is false": {
+			inOpts: RemoveEnvFromAppOpts{
+				App: &config.Application{
+					Name:      "phonetool",
+					AccountID: "1234",
+					Version:   "1",
+				},
+				EnvToDelete: &config.Environment{
+					Name:      "test",
+					AccountID: "1234",
+					Region:    "us-west-2",
+				},
+				Environments: []*config.Environment{
+					{
+						Name:      "test",
+						AccountID: "1234",
+						Region:    "us-west-2",
+					},
+					{
+						Name:      "prod",
+						AccountID: "1234",
+						Region:    "us-east-2",
+					},
+				},
+			},
+			mock: func(t *testing.T, ctrl *gomock.Controller) CloudFormation {
+				cfn := mocks.NewMockcfnClient(ctrl)
+				appStackSet := mocks.NewMockstackSetClient(ctrl)
+				s3 := mocks.NewMocks3Client(ctrl)
+				ecr := mocks.NewMockimageRemover(ctrl)
+				regionalCfn := mocks.NewMockcfnClient(ctrl)
+				// Empty ECR and S3
+				appStackSet.EXPECT().InstanceSummaries("phonetool-infrastructure", gomock.Any(), gomock.Any()).Return([]stackset.InstanceSummary{
+					{
+						Region:  "us-west-2",
+						StackID: "some-stack",
+					},
+				}, nil)
+				regionalCfn.EXPECT().Describe("some-stack").Return(mockValidAppResourceStack(), nil)
+				s3.EXPECT().EmptyBucket("tests3-bucket-us-west-2").Return(nil)
+				ecr.EXPECT().ClearRepository("phonetool-svc").Return(nil)
+				// Delete stackset instance
+				appStackSet.EXPECT().DeleteInstance("phonetool-infrastructure", "1234", "us-west-2").Return("123", nil)
+				appStackSet.EXPECT().WaitForOperation("phonetool-infrastructure", "123").Return(nil)
+
```

Comment:
nit: we can add ⬇️  to be super clear that we want no such call. The other maintainers would be able to see the intention clearly.
```go
appStackSet.Expect().Describe("phonetool-infrastructure").Times(0)
appStackSet.Expect(). Update("phonetool-infrastructure").Times(0)

```